bcda/main.go

```diff
@@ -379,3 +401,49 @@ func getEnvInt(varName string, defaultVal int) int {
 	}
 	return defaultVal
 }
+
+func removeExpired(hrThreshold int) {
+	fmt.Println("Cleaning up expired jobs...")
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	var jobs []models.Job
+	err := db.Find(&jobs, "status = ?", "Completed").Error
+	if err != nil {
+		log.Error(err)
+	}
+
+	expDir := os.Getenv("FHIR_EXPIRED_DIR")
+	if _, err = os.Stat(expDir); os.IsNotExist(err) {
+		err = os.MkdirAll(expDir, os.ModePerm)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
+	for _, j := range jobs {
+		t := j.CreatedAt
+		elapsed := time.Since(t).Hours()
+		if int(elapsed) >= hrThreshold {
+			j.Status = "Expired"
+			err = db.Save(j).Error
```

Comment:
I would just move this way down to the bottom of the code and then make sure the work succeeded before marking it as having succeeded.  Worst case there is you move the files, but leave the job as completed.

Reply:
Yes! I'll try this, hopefully this will satisfy the atomic operation problem.